msticpy/common/pkg_config.py

```diff
@@ -200,5 +204,178 @@ def _get_top_module():
+    if not mp_config and not config_file:
+        mp_config = settings
+
+    if not isinstance(mp_config, dict):
+        raise TypeError("Unknown format for configuration settings.")
+
+    mp_errors, mp_warn = _validate_azure_sentinel(mp_config=mp_config)
+
+    auth_key_providers = [
+        "OTX",
+        "VirusTotal",
+        "XForce",
+        "OpenPageRank",
+        "GeoIPLite",
+        "IPStack",
+    ]
+    for conf_section in ["TIProviders", "OtherProviders", "DataProviders"]:
+        prov_errors, prov_warn = _check_provider_settings(
+            mp_config=mp_config.get(conf_section, {}),
+            section=conf_section,
+            key_provs=auth_key_providers,
+        )
+        mp_errors.extend(prov_errors)
+        mp_warn.extend(prov_warn)
+
+    if "AzureCLI" not in mp_config:
+        mp_warn.append("No AzureCLI section in settings.")
+    else:
+        prov_errors, prov_warn = _check_provider_settings(
+            mp_config=mp_config, section=None, key_provs=None
+        )
+        mp_errors.extend(prov_errors)
+        mp_warn.extend(prov_warn)
+
+    _print_validation_report(mp_errors, mp_warn)
+    if mp_errors or mp_warn:
+        return mp_errors, mp_warn
+    return None
+
+
+def _print_validation_report(mp_errors, mp_warn):
+    if mp_errors:
+        title = "\nThe following configuration errors were found:"
+        print(title, "\n", "-" * len(title))
+        for err in mp_errors:
+            print(err)
+    else:
+        print("No errors found.")
+    if mp_warn:
+        title = "\nThe following configuration warnings were found:"
+        print(title, "\n", "-" * len(title))
+        for err in mp_warn:
+            print(err)
+    else:
+        print("No warnings found.")
+
+
+def _validate_azure_sentinel(mp_config):
+    mp_errors = []
+    as_settings = mp_config.get("AzureSentinel", {})
+    if not as_settings:
+        mp_errors.append("Missing or empty 'AzureSentinel' section")
+    ws_settings = as_settings.get("Workspaces", {})
+    if not ws_settings:
+        mp_errors.append("Missing or empty 'Workspaces' section in")
+    no_default = True
+    for ws, ws_settings in ws_settings.items():
+        if ws == "Default":
+            no_default = False
+        ws_id = ws_settings.get("WorkspaceId")
+        if not ws_id and not is_valid_uuid(ws_id):
+            mp_errors.append(f"Invalid GUID for WorkspaceId in {ws} section")
+        ten_id = ws_settings.get("TenantId")
+        if not ten_id and not is_valid_uuid(ten_id):
+            mp_errors.append(f"Invalid GUID for TenantId in {ws} section")
+    mp_warnings = ["No default workspace set"] if no_default else []
+    return mp_errors, mp_warnings
+
+
+def _check_provider_settings(mp_config, section, key_provs):
+    mp_errors = []
+    mp_warnings = []
+    if not mp_config:
+        mp_warnings.append(f"'{section}' section has no settings.")
+    for p_name, p_setting in mp_config.items():
+        sec_args = p_setting.get("Args")
```

Comment:
Throws an attribute error if you call validate_config() with config_file set. 
If config_file contains AzureCli and has an empty QueryDefinitions section this gets called with with ('QueryDefinitions', None) and tries get("Args")  on None (my config file was like this).
Could catch where p_setting is none and raise this as an mp_warning

Reply:
I think I've fixed this - it should have only been sending the AzureCLI section. I've pulled it out into a temp dictionary so that it matches the pattern of the TIProviders (i.e. how far down the dict tree the provider settings are).
If AzureCLI settings are in the DataProviders section this will have been processed already